factory/crypto/cryptoComponents.go

```diff
@@ -105,6 +119,7 @@ func NewCryptoComponentsFactory(args CryptoComponentsFactoryArgs) (*cryptoCompon
 		importModeNoSigCheck:                 args.ImportModeNoSigCheck,
 		enableEpochs:                         args.EnableEpochs,
 		noKeyProvided:                        args.NoKeyProvided,
+		p2pKeyPemFileName:                    args.P2pKeyPemFileName,
```

Comment:
perhaps add a check for `len(args.P2pKeyPemFileName) == 0`

Reply:
the check was not added intentionally, so that if there is no file provided, a new key will be generated

Reply:
oh, right